tests/upgrade/test_upgrade_from.sh

```diff
@@ -35,75 +35,67 @@ LOCAL_FORTIO_LOG=/tmp/fortio_local.log
-   cur_branch=`git rev-parse --abbrev-ref HEAD`
-   cur_sha=`git rev-parse HEAD`
+   original_branch=$(git rev-parse --abbrev-ref HEAD)
+   git checkout "tags/${1}" || die "Cannot git checkout, ensure your workspace is clean."
    writeMsg "Switching branch from ${original_branch} to tags/${1}"
 }
 
 gitRestorePrevBranch() {
-   git checkout ${original_branch}
-   cur_branch=`git rev-parse --abbrev-ref HEAD`
-   cur_sha=`git rev-parse HEAD`
+   git checkout "${original_branch}"
    writeMsg "Switching branch to ${original_branch}"
 }
 
 installIstioSystemAtTag() {
-   gitChangeToTag ${2}
-   installIstioSystemAtVersionHelmTemplate ${1} ${2}
+   gitChangeToTag "${2}"
+   installIstioSystemAtVersionHelmTemplate "${1}" "${2}"
    gitRestorePrevBranch
 }
 
 installTest() {
-    writeMsg "Installing test deployments"
-   kubectl apply -n ${TEST_NAMESPACE} -f ${ISTIO_ROOT}/tests/upgrade/templates/gateway.yaml
+   writeMsg "Installing test deployments"
+   kubectl apply -n "${TEST_NAMESPACE}" -f "${ISTIO_ROOT}/tests/upgrade/templates/gateway.yaml"
    # We don't want to auto-inject into istio-system, so echosrv and load client must be in different namespace.
-   kubectl apply -n ${TEST_NAMESPACE} -f ${ISTIO_ROOT}/tests/upgrade/templates/fortio.yaml
+   kubectl apply -n "${TEST_NAMESPACE}" -f "${ISTIO_ROOT}/tests/upgrade/templates/fortio.yaml"
    sleep 10
 }
 
 # Sends traffic from internal pod (Fortio load command) to Fortio echosrv.
 sendInternalRequestTraffic() {
    writeMsg "Sending internal traffic"
-   kubectl apply -n ${TEST_NAMESPACE} -f ${ISTIO_ROOT}/tests/upgrade/templates/fortio-cli.yaml
+   kubectl apply -n "${TEST_NAMESPACE}" -f "${ISTIO_ROOT}/tests/upgrade/templates/fortio-cli.yaml"
 }
 
 # Sends external traffic from machine test is running on to Fortio echosrv through external IP and ingress gateway LB.
 sendExternalRequestTraffic() {
     writeMsg "Sending external traffic"
-    fortio load -c 32 -t 300s -qps 10 -H "Host:echosrv.test.svc.cluster.local" http://${1}/echo?size=200 &> ${LOCAL_FORTIO_LOG}
+    fortio load -c 32 -t 300s -qps 10 -H "Host:echosrv.test.svc.cluster.local" "http://${1}/echo?size=200" &> "${LOCAL_FORTIO_LOG}"
 }
 
 restartDataPlane() {
     # Apply label within deployment spec.
     # This is a hack to force a rolling restart without making any material changes to spec.
-    cur_time=$(date +%s)
     writeMsg "Restarting deployment ${1}"
-    kubectl patch deployment ${1} -n ${TEST_NAMESPACE} -p'{"spec":{"template":{"spec":{"containers":[{"name":"echosrv","env":[{"name":"RESTART_","value":"1"}]}]}}}}'
+    kubectl patch deployment "${1}" -n "${TEST_NAMESPACE}" -p'{"spec":{"template":{"spec":{"containers":[{"name":"echosrv","env":[{"name":"RESTART_","value":"1"}]}]}}}}'
 }
 
 writeMsg() {
-    printf "\n\n****************\n\n${1}\n\n****************\n\n"
-
+    printf "\\n\\n****************\\n\\n%s\\n\\n****************\\n\\n" "${1}"
```

Comment:
change ${1} to ${@}
No quotes are needed when calling writeMsg()
for example:
       writeMsg no quotes is needed

Reply:
It seems it doesn't work. It adds line breaks to printf.

Reply:
it'll work with echo. 

But I think it's fine as is